mantidproject/mantid
# FindReflectometryLines version 2

**Description of work.**

This PR introduces version 2 of the `FindReflectometryLines` algorithm deprecating the old version. Version 2 is a complete rewrite implementing the peak fitting procedure from `LoadILLReflectometry` with some enhancements.

**Report to nobody.** Let's keep this one secret!

**To test:**

The algorithm is supposed to find the (fractional) workspace index of the peak center in a reflectometry dataset. The following script illustrates the usage with some unit test data:
```python
ws = LoadILLReflectometry('ILL/D17/317369.nxs')
ExtractMonitors(ws, DetectorWorkspace='ws')
pos = FindReflectometryLines('ws')
print(pos.LineCentre)
```

Check that the `pos.LineCentre` and the single valued workspace *out* contain the same fractional workspace index and that this corresponds to the peak centre in *314369*. There are other sample reflectometry files available in the unit/doc test data directories.

Try out the *RangeLower*, *RangeUpper*, *StartWorkspaceIndex* and *EndWorkspaceIndex* properties. Can you break the algorithm?

Implements the peak finding part of #22734.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
